single-solution/SolutionPage.js

```diff
@@ -76,16 +78,45 @@ function fill_fields(){
         var single_solution = SingleSolution(target_solution_concentration, total_volume,
             solute_molecular_weight);
 
-        /* Calculates the mass of solute to add to the solvent using the 'solid()' function within SingleSolution object.*/
-        var mass_of_solute_to_add = single_solution.solid();
+        
 
         /* Gets molarity variable from the solution concentration input field. (same as target concentration)*/
         var molarity = $("#solution_concentration").val();
 
+        var mass_of_solute_to_add;
+
+        if(page == "SOLID"){
+
+
+            /* Calculates the mass of solute to add to the solvent using the 'solid()' function within SingleSolution object.*/
+            mass_of_solute_to_add = single_solution.solid();
+            window.alert(mass_of_solute_to_add);
+            $("#massToAdd").val(mass_of_solute_to_add + "g"); // mass presented as 'grams'
+
+        }else if(page == "GRAV"){
+            window.alert(page);
+
+            mass_of_solute_to_add = single_solution.solid();
+            window.alert(mass_of_solute_to_add);
+            $("#massToAdd").val(mass_of_solute_to_add + "g"); // mass presented as 'grams'
+
+
+        }else if(page == "VOLU"){
+
+            var density = parseFloat($("#density").val());
+            window.alert(density);
+
+            mass_of_solute_to_add = single_solution.liquid.volume(density);
+            window.alert(mass_of_solute_to_add);
+
+            $("#massToAdd").val(mass_of_solute_to_add + " mL"); //answer presented in mL
+
+
+        }
+
```

Comment:
Above block of changes changes answer output based on what solution is being created.